crates/deltalake-core/src/operations/drop_constraints.rs

```diff
@@ -0,0 +1,247 @@
+    log_store: LogStoreRef,
+    /// Additional metadata to be added to commit
+    app_metadata: Option<HashMap<String, serde_json::Value>>,
+}
+
+impl DropConstraintBuilder {
+    /// Create a new builder
+    pub fn new(log_store: LogStoreRef, snapshot: DeltaTableState) -> Self {
+        Self {
+            name: None,
+            raise_if_not_exists: true,
+            snapshot,
+            log_store,
+            app_metadata: None,
+        }
+    }
+
+    /// Specify the constraint to be removed
+    pub fn with_constraint<S: Into<String>>(mut self, name: S) -> Self {
+        self.name = Some(name.into());
+        self
+    }
+
+    /// Specify if you want to raise if the constraint does not exist
+    pub fn with_raise_if_not_exists(mut self, raise: bool) -> Self {
+        self.raise_if_not_exists = raise;
+        self
+    }
+
+    /// Additional metadata to be added to commit info
+    pub fn with_metadata(
+        mut self,
+        metadata: impl IntoIterator<Item = (String, serde_json::Value)>,
+    ) -> Self {
+        self.app_metadata = Some(HashMap::from_iter(metadata));
+        self
+    }
+}
+
+impl std::future::IntoFuture for DropConstraintBuilder {
+    type Output = DeltaResult<DeltaTable>;
+
+    type IntoFuture = BoxFuture<'static, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        let mut this = self;
+
+        Box::pin(async move {
+            let name = match this.name {
+                Some(v) => v,
+                None => return Err(DeltaTableError::Generic("No name provided".to_string())),
+            };
+
+            let mut metadata = this.snapshot.metadata().clone();
+            let configuration_key = format!("delta.constraints.{}", name);
+
+            let found_constraint = metadata.configuration.contains_key(&configuration_key);
+            if this.raise_if_not_exists && !found_constraint {
+                return Err(DeltaTableError::Generic(format!(
+                    "Constraint with name: {} doesn't exists",
+                    name
+                )));
+            } else if !this.raise_if_not_exists && !found_constraint {
+                return Ok(DeltaTable::new_with_state(this.log_store, this.snapshot));
+            }
+
+            metadata
+                .configuration
+                .remove(format!("delta.constraints.{}", name).as_str());
+
+            let mut contains_constraints: bool = false;
+            for key in metadata.configuration.keys() {
+                if key.starts_with("delta.constraints") {
+                    contains_constraints = true;
+                    break;
+                } else {
+                    continue;
+                }
+            }
+
+            let operational_parameters = HashMap::from_iter([("name".to_string(), json!(&name))]);
+
+            let operations = DeltaOperation::DropConstraint { name: name.clone() };
+
+            let app_metadata = match this.app_metadata {
```

Comment:
`.unwrap_or_default()` instead

Reply:
Ah of course there is something for that :P Thanks!